scripts/scan-txlog/unclaimed_sv_rewards.py

```diff
@@ -0,0 +1,697 @@
+                LOG.debug(f"Adding closing round {closing_round} to active_closing_rounds")
+                self.active_closing_rounds_contracts[event.contract_id] = round_number
+                self.active_closing_rounds[round_number] = closing_round
+
+
+    def process_exercised_event(self, transaction: TransactionTree, event: ExercisedEvent):
+        match event.template_id.qualified_name:
+            case TemplateQualifiedNames.sv_reward_coupon:
+                match event.choice_name:
+                    case "SvRewardCoupon_DsoExpire":
+                        self.handle_sv_reward_coupon_exercise(transaction, event, RewardStatus.EXPIRED)
+                    case "SvRewardCoupon_ArchiveAsBeneficiary":
+                        self.handle_sv_reward_coupon_exercise(transaction, event, RewardStatus.CLAIMED)
+                    case choice if event.is_consuming:
+                        self._fail(
+                            f"Unexpected consuming choice {choice} on "
+                            f"{TemplateQualifiedNames.sv_reward_coupon} — aborting."
+                        )
+                    case _:
+                        self.process_events(transaction, event.child_event_ids)
+            case TemplateQualifiedNames.issuing_mining_round:
+                match event.choice_name:
+                    case "Archive":
+                        match self.active_issuing_rounds_contracts.pop(event.contract_id, None):
+                            case None:
+                                pass
+                            case round_number:
+                                self.active_issuing_rounds.pop(round_number)
+                    case _:
+                        self.process_events(transaction, event.child_event_ids)
+            case TemplateQualifiedNames.closing_mining_round:
+                match event.choice_name:
+                    case "Archive":
+                        match self.active_closing_rounds_contracts.pop(event.contract_id, None):
+                            case None:
+                                pass
+                            case round_number:
+                                self.active_closing_rounds.pop(round_number)
+                    case _:
+                        self.process_events(transaction, event.child_event_ids)
+            case _:
+                self.process_events(transaction, event.child_event_ids)
+
+
+    def handle_sv_reward_coupon_exercise(self, transaction, event, status):
+        # Only handle SvRewardCoupons created within the configured time range for the given beneficiary
+        match self.active_rewards.pop(event.contract_id, None):
+            # None means either:
+            # - The reward was created outside the time range, or
+            # - The exercise event is for a different beneficiary
+            case None:
+                pass
+            case reward:
+                issuing = self.active_issuing_rounds.get(reward.round)
```

Comment:
can we move the round lookup into the `match status` part? It seems like then you know which one you need: For claimed you lookup the issuing round, for expired you lookup the closed round.